src/services/offer.js

```diff
@@ -155,6 +156,33 @@ class OfferService {
         return offer;
     }
 
+    blockByCompany(owner) {
+        return Offer.updateMany(
+            { owner, isHidden: false },
+            {
+                isHidden: true,
+                hiddenReason: HiddenOfferReasons.COMPANY_BLOCKED,
+            },
+            (err) => {
+                if (err) {
+                    throw err;
+                }
+            });
+    }
+
+    unblockByCompany(owner) {
+        return Offer.updateMany(
+            { owner, isHidden: true, hiddenReason: HiddenOfferReasons.COMPANY_BLOCKED },
+            {
+                isHidden: false,
+                $unset: { hiddenReason: undefined, adminReason: undefined },
+            },
+            (err) => {
+                if (err) {
+                    throw err;
+                }
+            });
```

Comment:
doesn't this work naturally without passing the callback? I think it behaves like an async function which will naturally reject on error, can you try it? To force an error maybe you can try passing a wrong field on the update object, something like `$foo: "bar"` (?)

Reply:
https://mongoosejs.com/docs/faq.html#duplicate-query as per this, I am 99% sure that calling .updateMany without the callback will execute as a promise, so you can remove this part (and in the "block" part as well